### ISSUE TITLE
Timestamps fields

### ISSUE THREAD

**Issue by vfreitas-** [scottwrobinson/camo#40]
Maybe add the timestamps fields by default?(created_at, updated_at).

**Comment by michaeljota** [comment]
:+1: +1

**Comment by scottwrobinson** [comment]
I'll consider it. I hesitate to add data by default to people's models, but
maybe it can be an option.

I'll leave this open for discussion for a bit to gauge interest.

Thanks!

Scott

**Comment by vfreitas-** [comment]
You can have a boolean variable, that will decide if the timestamp fields will be added.

`class User extends Document {
   constructor() {



}`

**Comment by vfreitas-** [comment]
You can have a boolean variable, that will decide if the timestamp fields will be added.

```
class User extends Document {
	constructor() {
		//... 

		this.timestamps = false; //default = true
	
		//...
	}
}
```

**Comment by michaeljota** [comment]
Maybe not to be enable by default, but that it's a way to go.

**Comment by scottwrobinson** [comment]
Okay so I'm finally getting back to this. Thinking about it some more, I think it would be a nice inclusion, just not enabled by default. I believe features like this should be opt-in instead of opt-out.

The `timestamps` property would likely have to be set using a static method. Like this:

```javascript
class User extends Document {
    static timestamps() {
        return true;
    }
}
```

You could also return names of the created_at/updated_at fields for more customization:

```javascript
class User extends Document {
    static timestamps() {
        return {
            created_at: 'createdTime',
            updated_at: 'updatedTime'
        };
    }
}
```

Doing it in the constructor, as shown by @vfreitas-, wouldn't work because then we couldn't access the `timestamps` property in any static methods (like `User.findOneAndUpdate()`).

Thoughts?

**Comment by samirbr** [comment]
What about use a timestamp decorator?

```
@timestamp
class User extends Document {
}
```

You could pass timestamp field names too:

```
@timestamp({
  created_at: 'createdTime',
  updated_at: 'updatedTime'
})
class Use extends Document {
}
```

**Comment by michaeljota** [comment]
I don't think ES6 have decorators. For what I understand after googling,
decorators are suppose to come in 2016 with ES2016(ES7).

However, for a long proposal that would be a good idea.

2016-05-31 19:51 GMT-04:00 Samir El Aouar <notifications@github.com>:

**Comment by scottwrobinson** [comment]
Yep, ES6 does not have decorators. As nice as this feature would be, we'd
then have to require application code to use a transpiler like Babel, which
does not work with Camo's goal.

I'd like to implement this feature in ES6 soon, but if anyone would like to
submit a PR for it sooner, I'd be happy to accept.

**Comment by devdebonair** [comment]
Was a pull request submitted for this yet?
